pandas/core/indexes/multi.py

```diff
@@ -2178,6 +2209,47 @@ def append(self, other):
         except (TypeError, IndexError):
             return Index(new_tuples)
 
+    def get_index_data(self, data_index, column_name, other, search_self=False):
+
+        # Returns original data if the data_index input has data for this column name
+        if column_name in data_index.names:
+            Index_position = data_index.names.index(column_name)
+            data = data_index._get_level_values(Index_position)
+            return data
+
+        else:
+
+            # If the data_index input is from other and if it don't
+            # have the column name, it returns an Index filled with pd.NA
+            # with data type that the other dataframe has the column.
+            if search_self is True:
+                if column_name in self.names:
+                    Index_position = self.names.index(column_name)
+                    NA_type = self.levels[Index_position].dtype
+                    data = Index([NA] * data_index.size, dtype=NA_type)
+                    return data
+
+            for o in other:
+                if o is not data_index and column_name in o.names:
+                    Index_position = o.names.index(column_name)
+                    NA_type = o.levels[Index_position].dtype
+                    data = Index([NA] * data_index.size, dtype=NA_type)
+                    return data
+
```

Comment:
This entire function is to return the data that the index have. if the MultiIndex does not contain the column, it would return a ```pd.Index``` filled with ```pd.NA```. Example,

```
MultiIndex0 = pd.MultiIndex( Index(data=[ 0 , 1 ], name="Stationary"), Index(data=[ 2 , 3 ], name="Tree"))
MultiIndex1 = pd.MultiIndex( Index(data=[ 4 , 5 ], name="Food"), Index(data=[ 6 , 7 ], name="Tree"))
MultiIndex2 =  pd.MultiIndex( Index(data=[ 8 , 9 ], name="Stationary"), Index(data=[ 10 , 11 ], name="Tree"))
self = DataFrame0
other = [DataFrame1, DataFrame2]

self. get_index_data( data_index=MultiIndex2,
                            column_name= "Food",
                            other=other,
                            search_self=True ) 

self. get_index_data( data_index=self,
                            column_name= "Stationary",
                            other=other,
                            search_self=False ) 
```
would yield
```
pd.Index([ pd.NA, pd.NA ])

and 

pd.Index([ 0 , 1 ])